subsys/llext/llext_load.c

```diff
@@ -186,7 +186,8 @@ static int llext_find_tables(struct llext_loader *ldr)
 
 	if (!ldr->sects[LLEXT_MEM_SHSTRTAB].sh_type ||
 	    !ldr->sects[LLEXT_MEM_STRTAB].sh_type ||
-	    !ldr->sects[LLEXT_MEM_SYMTAB].sh_type) {
+	    !ldr->sects[LLEXT_MEM_SYMTAB].sh_type ||
+	    sect_cnt != 3) {
```

Comment:
IIRC this check wasn't needed before because the loop was only running up to three times:
```
	for (i = 0, sect_cnt = 0, pos = ldr->hdr.e_shoff;
	       i < ldr->hdr.e_shnum && sect_cnt < 3;
	       i++, pos += ldr->hdr.e_shentsize) {
```
so, yes, we can either restore it there or add a check here. The former would find the problem sooner and would localise the use of `sect_cnt` to the loop :-)